### PR TITLE
feat: add a temporary reset script for the local network

### PR DESCRIPTION
**Pull request by antouhou**
## Issue being fixed or feature implemented

Add a temporary reset script, so all the devs have the same script.
*TO BE REMOVED AFTER RESET COMMAND REVISION*

## What was done?

Add a simple reset script

## How Has This Been Tested?

Run on the machine

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: bin/temporary_reset_script.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  for (( NODE_INDEX=1; NODE_INDEX<=MASTERNODES_COUNT; NODE_INDEX++ ))
+  do
+      ./bin/dashmate config:set --config=${CONFIG_NAME}_${NODE_INDEX} core.miner.interval "${MINING_INTERVAL_IN_SECONDS}s"
+  done
+./bin/dashmate config:set --config=${CONFIG_NAME}_seed core.miner.address $FAUCET_ADDRESS
+
+echo "Sending 1000 tDash to the ${FAUCET_ADDRESS} for tests"
+./bin/dashmate wallet:mint 1000 --config=${CONFIG_NAME}_1 --address=${FAUCET_ADDRESS}
+
+if [ $BUILD_DAPI_AFTER_SETUP == true ]
```
Comment (shumkov) [inline bin/temporary_reset_script.sh:58]: What is the point to set it after setup?

File: bin/temporary_reset_script.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+if [ $BUILD_DAPI_BEFORE_SETUP == true ]
+then
+  echo "Setting dapi build directory before the setup"
+  ./bin/dashmate config:set --config=${CONFIG_NAME} platform.dapi.api.docker.build.path $DAPI_REPO_PATH
+fi
+
+./bin/dashmate setup ${CONFIG_NAME} -v --node-count=${MASTERNODES_COUNT}
+
+echo "Setting development environment"
+./bin/dashmate config:set --config=${CONFIG_NAME} environment development
```
Comment (shumkov) [inline bin/temporary_reset_script.sh:41]: It's already development for local

File: bin/temporary_reset_script.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+./bin/dashmate config:set --config=${CONFIG_NAME} platform.drive.abci.log.stdout.level trace
+
+# Set up a miner for tests
+echo "Enabling miner in config with an interval of ${MINING_INTERVAL_IN_SECONDS} seconds"
+./bin/dashmate config:set --config=${CONFIG_NAME}_seed core.miner.enable true
+./bin/dashmate config:set --config=${CONFIG_NAME}_seed core.miner.interval "${MINING_INTERVAL_IN_SECONDS}s"
+  for (( NODE_INDEX=1; NODE_INDEX<=MASTERNODES_COUNT; NODE_INDEX++ ))
+  do
+      ./bin/dashmate config:set --config=${CONFIG_NAME}_${NODE_INDEX} core.miner.interval "${MINING_INTERVAL_IN_SECONDS}s"
+  done
```
Comment (shumkov) [inline bin/temporary_reset_script.sh:50]: We don't need this with https://github.com/dashevo/dashmate/pull/360

Comment (antouhou) [reply bin/temporary_reset_script.sh:50]: It's not merged yet though. Do you want to merge #360 first?

Comment (shumkov) [reply bin/temporary_reset_script.sh:50]: It's merged. Now we have -d and -m flags so you can remove unnecessary configuration from here.

File: bin/temporary_reset_script.sh [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+BUILD_DRIVE=false
+
+CONFIG_NAME="local"
+
+MASTERNODES_COUNT=3
+
+echo "Removing all docker containers and volumes..."
+docker rm -f -v $(docker ps -a -q); docker volume prune -f; rm -rf ~/.dashmate/
+
+./bin/dashmate config:set --config=${CONFIG_NAME} core.debug 1
```
Comment (shumkov) [inline bin/temporary_reset_script.sh:24]: Just pass -d to setup command

File: bin/temporary_reset_script.sh [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+./bin/dashmate config:set --config=${CONFIG_NAME}_seed core.miner.enable true
+./bin/dashmate config:set --config=${CONFIG_NAME}_seed core.miner.interval "${MINING_INTERVAL_IN_SECONDS}s"
+  for (( NODE_INDEX=1; NODE_INDEX<=MASTERNODES_COUNT; NODE_INDEX++ ))
+  do
+      ./bin/dashmate config:set --config=${CONFIG_NAME}_${NODE_INDEX} core.miner.interval "${MINING_INTERVAL_IN_SECONDS}s"
+  done
+./bin/dashmate config:set --config=${CONFIG_NAME}_seed core.miner.address $FAUCET_ADDRESS
+
+echo "Sending 1000 tDash to the ${FAUCET_ADDRESS} for tests"
+./bin/dashmate wallet:mint 1000 --config=${CONFIG_NAME}_1 --address=${FAUCET_ADDRESS}
```
Comment (shumkov) [inline bin/temporary_reset_script.sh:54]: I would use the seed node for that.

File: bin/temporary_reset_script.sh (hunk 3 above)
Comment (antouhou) [reply bin/temporary_reset_script.sh]: Done

File: bin/temporary_reset_script.sh (hunk 4 above)
Comment (antouhou) [reply bin/temporary_reset_script.sh]: Done

File: bin/temporary_reset_script.sh [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+if [ $BUILD_DAPI_BEFORE_SETUP == true ]
+then
+  echo "Setting dapi build directory before the setup"
+  ./bin/dashmate config:set --config=${CONFIG_NAME} platform.dapi.api.docker.build.path $DAPI_REPO_PATH
+fi
+
+./bin/dashmate setup ${CONFIG_NAME} -v --node-count=${MASTERNODES_COUNT}
+
+echo "Setting log levels to trace"
+./bin/dashmate config:set --config=${CONFIG_NAME} platform.drive.abci.log.stdout.level trace
```
[the remark this answers is not in the document]
Comment (antouhou) [reply bin/temporary_reset_script.sh:40]: Done

File: bin/temporary_reset_script.sh [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+# Set up a miner for tests
+echo "Enabling miner in config with an interval of ${MINING_INTERVAL_IN_SECONDS} seconds"
+./bin/dashmate config:set --config=${CONFIG_NAME}_seed core.miner.enable true
+./bin/dashmate config:set --config=${CONFIG_NAME}_seed core.miner.interval "${MINING_INTERVAL_IN_SECONDS}s"
+  for (( NODE_INDEX=1; NODE_INDEX<=MASTERNODES_COUNT; NODE_INDEX++ ))
+  do
+      ./bin/dashmate config:set --config=${CONFIG_NAME}_${NODE_INDEX} core.miner.interval "${MINING_INTERVAL_IN_SECONDS}s"
+  done
+./bin/dashmate config:set --config=${CONFIG_NAME}_seed core.miner.address $FAUCET_ADDRESS
```
[the remark this answers is not in the document]
Comment (antouhou) [reply bin/temporary_reset_script.sh]: Removed this line altogether

File: bin/temporary_reset_script.sh (hunk 5 above)
Comment (antouhou) [reply bin/temporary_reset_script.sh]: Done

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**